### PR TITLE
SALTO-5981: Change dependency of object types

### PR DESCRIPTION
**Pull request by IdoZakk**
Reversed the dependency of object type to its non salto parent (appears in a referenced field) on removal, so it will behave the same as salto parents. I also added ignore to 404 on removal.

---

See the Jira ticket. This is a bit of a patch, the full solution is probably to make the salto parent to be the same.

---
_Release Notes_: 
Jira adapter:
* Deployment of object types will not cause circular dependency

---
_User Notifications_: 
None

### CODE REVIEW

**Review by kami-delaroz — APPROVED** [review]
🔥 🔥 🔥
 Great!

File: packages/jira-adapter/src/dependency_changers/index.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ import { fieldConfigurationDependencyChanger } from './field_configuration'
 import { jsmProjectToJsmFieldDependencyChanger } from './jsm_project_to_jsm_field'
 import { rootObjectTypeToObjectSchemaDependencyChanger } from './root_object_type_to_schema'
 import { issueLayoutDependencyChanger } from './issue_layout_dependency'
+import { objectTypeParentReversalDependencyChanger } from './object_type_parent_reversal'
```
[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/dependency_changers/index.ts:30]: I am confused. The file name also contains the reversal suffix

[the remark this answers is not in the document]
Comment (kami-delaroz) [reply packages/jira-adapter/src/dependency_changers/index.ts:30]: I meant other file names didn't have reversal suffix, even in similar use cases

[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/dependency_changers/index.ts]: Oh, it seems more informative

[the remark this answers is not in the document]
Comment (kami-delaroz) [reply packages/jira-adapter/src/dependency_changers/index.ts]: I don't have a strong opinion here, just that the rest file names are according to this specific convention, for better or worse :)

[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/dependency_changers/index.ts]: Sorry, it was a half baked comment that got submitted by mistake. I fixed it

**Pull request merged**